porting/index.md

```diff
@@ -1,5 +1,12 @@
 # Porting guide
 
+## Device considerations
+
+* **Kernel:** Halium currently requires a device with a kernel greater or equal version 3.1.0 - older device kernels are not compatible with the glibc build in the root filesystem being used. Check your device´s kernel version by using adb with uname -a for example.
+* **RAM:** While 1GB of RAM is sufficient to start the OS, it is recommended to have 2GB at least to have a good end user experience.
+* **Chipset:** Try to avoid Mediatek chipsets, they are not open-sourced and so there are nearly no usable Android trees for building available.
```

Comment:
> they are not open-sourced

The vendor is legally obligated to and also responsible for publishing the kernel sources and not Mediatek.

I am not really familiar with Mediatek devices so I don't know if there are any other parts needed to make a stable Android ROM but as there ARE stable Android ROMs for Mediatek devices, I don't believe that's the case. Yes, Qualcomm has CodeAurora Forum (=caf) where they publish complete Android ROM sources but I'm not sure how that helps Halium.
And unfortunately Qualcomm has a big monopoly on the higher-priced smartphone market and unfortunately they also like to abuse their power. I am also not familiar with higher-priced and newer Mediatek chipsets (like the Helio X30) so I can't compare it in performance with Qualcomm chips. Other chip vendors are mostly dead in the smartphone market anyways (except maybe HiSilicon (Kirin) and Samsung (Exynos)).

Reply:
Well I can just tell you that I am in contact with the BQ guys (at the CEO level :)) and what I am hearing from them is that they do not believe we can upgrade their Mediatek devices to 16.04 without the help of Mediatek. So, I assume that they deliver only the kernel parts etc. that they must according to GPL, but that these sources will not form a buildable tree. But let´s wait until I get more feedback from them. Maybe the wording needs to be different, still I think Mediatek is not a good option for us.

Reply:
Interesting. But actually I believe the GPL says that the sources must be compile-able and if a custom toolchain is used also build instructions for that. But yes, Mediatek cares *less* about what people do with their already-purchased devices than Qualcomm.

Reply:
So basically its like the following:

* Kernel sources are under GPL and must be released in full source by Mediatek
* Contents of the device and vendor folder are Apache-licensed, and do not need to be published in source-code form.
* BQ and other vendors accept an NDA plus probably pay a fee to get the full tree to build / customize. But they are not allowed to release that tree officially.
* Leaked versions of various trees are circling the Internet, but we should not use them, even if we find some for Mediatek devices. Mostly you never know if it has been modified or not, and then we do not want to have a lawsuit upon us.
* Because of that, Mediatek devices in general are full of security flaws and bugs, and probably are not the solid base that an alternative OS should be based on.

I really would write the note as I suggested. What does @bhush9 think?

Reply:
I don't think Mediatek is responsible for releasing the kernel sources but the phone manufacturers. And yes, I know that the rest of everything is not GPL licensed and therefore isn't required to be published.

Reply:
Ok there is no difference if it is the responsibility of the SoC vendor or the assembly company of the phone. It will still be an incomplete tree since they don´t need to release all parts. So why waste time with these phones in Halium?